Correlations: Fix incorrect state transitions in transformations editor

ifrost opened the pull request:
Fixes a few issues related to transformations editor. Please check #75223 for more details and reproduction steps.

Fixes #75223

- Editing transformation typewas not possible because of `<Select value={fieldVal.type} />`. This will always get the `default` value from the hook-form because `FieldArray` is not controlling the value. 
- To solve the above we could use `<Select value={field.value} ` from the `<Controller/>`. This, however works only after the page is rendered and will revert to defaults from the form after pressing "back" + "next" button (pressing "back" + "next"  causes the form to re-render and re-use defaults again)
- To make it work with "back"/"next" buttons we need to watch form state for changes and apply the value from the state. This is less performant but should be good enough in our case.
- To allow watching state of a values that are in an array I have extracted a row to a separate component to avoid calling `useWatch` inside a callback (that would break deleting rows).

ifrost committed:
Extract transformations editor row to a seprate component

ifrost committed:
Merge branch 'main' into ifrost/correlations-form-fix

ifrost committed:
Use css object instead of string literals

ifrost committed:
Update .betterer.results

gelicia reviewed (approved):
LGTM!

harisrozajac commented on public/app/features/correlations/Forms/TransformationsEditor.tsx:
```diff
@@ -26,21 +15,14 @@ const getStyles = (theme: GrafanaTheme2) => ({
     font-size: ${theme.typography.h5.fontSize};
```
can we put these into an object to avoid adding to the betterer warnings? something like: 
```
const getStyles = (theme: GrafanaTheme2) => {
  return {
    heading: css({
      fontSize: `${theme.typography.h5.fontSize}`,
      fontWeight: `${theme.typography.fontWeightRegular}`,
    }),
  };
};
```

ifrost replied:
Good point! Updated in c96d238d6f9e5de7d694178c30c3c947d8b3cac1

ifrost committed:
Merge branch 'main' into ifrost/correlations-form-fix

# Conflicts:
#	public/app/features/correlations/Forms/TransformationsEditor.tsx

ifrost committed:
Post merge fixes

ifrost committed:
Bring back validation rules

ifrost committed:
Switch to css/object

ifrost commented:
@gelicia / @harisrozajac I've resolved conflicts, brought back validation rules for type input (I'd removed them by accident 😨 ) and removed css literal strings.

ifrost committed:
Merge branch 'main' into ifrost/correlations-form-fix

# Conflicts:
#	public/app/features/correlations/Forms/TransformationsEditor.tsx

ifrost committed:
Post-merge fixes

Ensuring Stack from grafana-ui is used after conflicts with 25779bb6e5cedef712db15b9cc77ab4af68c31dd
